superset/assets/src/explore/components/controls/DateFilterControl.jsx

```diff
@@ -242,11 +247,32 @@ export default class DateFilterControl extends React.Component {
   }
 
   handleClick(e) {
+    const target = e.target;
     // switch to `TYPES.CUSTOM_START_END` when the calendar is clicked
-    if (this.startEndSectionRef && this.startEndSectionRef.contains(e.target)) {
+    if (this.startEndSectionRef && this.startEndSectionRef.contains(target)) {
       this.setTypeCustomStartEnd();
     }
+
+    // if user click outside popover, popover will hide and we will call onCloseDateFilterControl,
+    // but popover overlay trigger button is inside popover DOM h
```

Comment:
looks like this comment didn't get finished?

Reply:
:sweat: fixed, and added more comments.